x/btcstaking/keeper/hooks.go

```diff
@@ -0,0 +1,39 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	bbntypes "github.com/babylonchain/babylon/types"
+	"github.com/babylonchain/babylon/x/finality/types"
+)
+
+var _ types.FinalityHooks = Hooks{}
+
+// Hooks wrapper struct for BTC staking keeper
+type Hooks struct {
+	k Keeper
+}
```

Comment:
```suggestion
type Hooks struct {
	k *Keeper
}
```

to avoid unexpected copy of the keeper object?

Reply:
This is simply a wraper which will return the keeper when [Hooks()](https://github.com/babylonchain/babylon/blob/1f5671b25a2cb54a1cc75118d98956d1f716cf90/x/btcstaking/keeper/hooks.go#L20) is called. No values a copied. All of our modules and cosmos-sdk modules are doing it the same way